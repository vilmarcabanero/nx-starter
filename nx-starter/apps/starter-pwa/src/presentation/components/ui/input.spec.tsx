import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Input } from './input';

describe('Input Component Snapshots', () => {
  describe('input types', () => {
    it('renders default text input', () => {
      const { container } = render(<Input />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders text input with placeholder', () => {
      const { container } = render(<Input placeholder="Enter text..." />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders email input', () => {
      const { container } = render(
        <Input type="email" placeholder="Enter email..." />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders password input', () => {
      const { container } = render(
        <Input type="password" placeholder="Enter password..." />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders number input', () => {
      const { container } = render(
        <Input type="number" placeholder="Enter number..." />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders search input', () => {
      const { container } = render(
        <Input type="search" placeholder="Search..." />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders file input', () => {
      const { container } = render(<Input type="file" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('input states', () => {
    it('renders disabled input', () => {
      const { container } = render(
        <Input disabled placeholder="Disabled input" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders readonly input', () => {
      const { container } = render(<Input readOnly value="Read only value" />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders required input', () => {
      const { container } = render(
        <Input required placeholder="Required field" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders input with value', () => {
      const { container } = render(<Input value="Input with value" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('input attributes', () => {
    it('renders input with custom id', () => {
      const { container } = render(
        <Input id="custom-id" placeholder="Custom ID" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders input with name attribute', () => {
      const { container } = render(
        <Input name="username" placeholder="Username" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders input with max length', () => {
      const { container } = render(
        <Input maxLength={50} placeholder="Max 50 characters" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders input with min/max for numbers', () => {
      const { container } = render(
        <Input type="number" min={0} max={100} placeholder="0-100" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('customization', () => {
    it('renders input with custom className', () => {
      const { container } = render(
        <Input className="custom-input" placeholder="Custom styling" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders input with aria-invalid', () => {
      const { container } = render(
        <Input aria-invalid="true" placeholder="Invalid input" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders input with aria-describedby', () => {
      const { container } = render(
        <Input aria-describedby="help-text" placeholder="With help text" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('input combinations', () => {
    it('renders required email input with custom class', () => {
      const { container } = render(
        <Input
          type="email"
          required
          className="w-full"
          placeholder="Required email"
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders disabled password input', () => {
      const { container } = render(
        <Input type="password" disabled placeholder="Disabled password" />
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders file input with multiple attribute', () => {
      const { container } = render(<Input type="file" multiple />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
