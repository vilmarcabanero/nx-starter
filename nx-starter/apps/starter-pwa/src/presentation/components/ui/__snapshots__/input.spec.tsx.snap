// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input Component Snapshots > customization > renders input with aria-describedby 1`] = `
<input
  aria-describedby="help-text"
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="With help text"
/>
`;

exports[`Input Component Snapshots > customization > renders input with aria-invalid 1`] = `
<input
  aria-invalid="true"
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Invalid input"
/>
`;

exports[`Input Component Snapshots > customization > renders input with custom className 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive custom-input"
  data-slot="input"
  placeholder="Custom styling"
/>
`;

exports[`Input Component Snapshots > input attributes > renders input with custom id 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  id="custom-id"
  placeholder="Custom ID"
/>
`;

exports[`Input Component Snapshots > input attributes > renders input with max length 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  maxlength="50"
  placeholder="Max 50 characters"
/>
`;

exports[`Input Component Snapshots > input attributes > renders input with min/max for numbers 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  max="100"
  min="0"
  placeholder="0-100"
  type="number"
/>
`;

exports[`Input Component Snapshots > input attributes > renders input with name attribute 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  name="username"
  placeholder="Username"
/>
`;

exports[`Input Component Snapshots > input combinations > renders disabled password input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  disabled=""
  placeholder="Disabled password"
  type="password"
/>
`;

exports[`Input Component Snapshots > input combinations > renders file input with multiple attribute 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  multiple=""
  type="file"
/>
`;

exports[`Input Component Snapshots > input combinations > renders required email input with custom class 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive w-full"
  data-slot="input"
  placeholder="Required email"
  required=""
  type="email"
/>
`;

exports[`Input Component Snapshots > input states > renders disabled input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  disabled=""
  placeholder="Disabled input"
/>
`;

exports[`Input Component Snapshots > input states > renders input with value 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  value="Input with value"
/>
`;

exports[`Input Component Snapshots > input states > renders readonly input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  readonly=""
  value="Read only value"
/>
`;

exports[`Input Component Snapshots > input states > renders required input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Required field"
  required=""
/>
`;

exports[`Input Component Snapshots > input types > renders default text input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
/>
`;

exports[`Input Component Snapshots > input types > renders email input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Enter email..."
  type="email"
/>
`;

exports[`Input Component Snapshots > input types > renders file input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  type="file"
/>
`;

exports[`Input Component Snapshots > input types > renders number input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Enter number..."
  type="number"
/>
`;

exports[`Input Component Snapshots > input types > renders password input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Enter password..."
  type="password"
/>
`;

exports[`Input Component Snapshots > input types > renders search input 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Search..."
  type="search"
/>
`;

exports[`Input Component Snapshots > input types > renders text input with placeholder 1`] = `
<input
  class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
  data-slot="input"
  placeholder="Enter text..."
/>
`;
